pkg/proxy/roundrobin.go

```diff
@@ -210,6 +221,10 @@ func (lb *LoadBalancerRR) OnUpdate(endpoints []api.Endpoints) {
 	for _, endpoint := range endpoints {
 		existingEndpoints, exists := lb.endpointsMap[endpoint.Name]
 		validEndpoints := filterValidEndpoints(endpoint.Endpoints)
+		// Need to compare sorted endpoints here, since they are shuffled below
+		// before being put into endpointsMap
+		sort.Strings(existingEndpoints)
```

Comment:
I am pretty sure that, given the reference-nature of slices, this will actually sort the lb.endpointsMap[endpoint.Name] slice, thus undoing your shuffle.

If we keep this general approach, you need to sort-and-copy the slice.